### PR TITLE
Set number of cpu to 255

### PR DESCRIPTION
**Pull request by miguelinux**
The max CPU number for a CC was 8, this number is not enough for some task.

Set number of cpu to 255

### CODE REVIEW

**Comment by mcastelino** [comment]
@miguelinux QEMU goes up to 255 CPU's and we have platforms with more than 32 CPUs. Even at 8k per vCPU we are looking at 2MB for 255 CPUs. I feel we should match the QEMU max number.

What was the measured overhead of each CPU.

/cc @grahamwhaley @sameo

File: kernel/kernel-config-4.9.x [hunk 1]
```diff
@@ -2268,7 +2268,7 @@ CONFIG_CRYPTO_SEQIV=y
 CONFIG_CRYPTO_CBC=y
 CONFIG_CRYPTO_CTR=y
 # CONFIG_CRYPTO_CTS is not set
-# CONFIG_CRYPTO_ECB is not set
+CONFIG_CRYPTO_ECB=y
```
Comment (gorozco1) [inline kernel/kernel-config-4.9.x:2271]: is this needed?

**Review by gorozco1 — APPROVED** [review]
lgtm, just a question about a setting

**Comment by gorozco1** [comment]
@miguelinux you need to push this again.

**Comment by grahamwhaley** [comment]
@mcastelino I too was thinking that we might not want to restrict this value - given the CC default setting for vcpu's is `-1`, which means 'same as host`, then I agree we should go for the QEMU max of 255 - unless there really is a huge overhead.
@miguelinux - do you know what the per-cpu overhead is?

**Comment by miguelinux** [comment]
@mcastelino, @grahamwhaley I don't known the overhead, but

Do you know a use case with a VM container use more than 64 cores?

64 cores does not bring a lot overhead as is the size of the word.

**Comment by jodh-intel** [comment]
I'm not sure I'm reading `kernel/smp.c` right, but I _think_ we could set `CONFIG_NR_CPUS` to a high value (say `255`). But we could potentially affect the amount of memory consumed by booting with:

- `nr_cpus = X` to restrict the total number of CPUs the kernel will (ever?) deal with.
- `maxcpus = Y` to restrict the number of cpus the kernel will bring online *at boot time*.

... with `nr_cpus - maxcpus` being the number of available hot-pluggable cpus.

But we'll need a kernel guru to confirm this ;)

/me notices that his Fedora box sets `CONFIG_NR_CPUS=8192` for the host kernel.

**Comment by grahamwhaley** [comment]
@jodh-intel I think ... you might be right - good thought. My guess is that it will still cost us a little, but not as much as if we just hard wire to a large value.
This needs a few measurements I feel - we could go with an interim of just wiring to the 255 and then at least opening an Issue to note we need to measure and maybe be more dynamic...

**Comment by gorozco1** [comment]
@miguelinux this got caught by the travis bug could you do a update in your PR please?

**Comment by jcvenegas** [comment]
any update on this PR?

**Comment by devimc** [comment]
https://github.com/clearcontainers/packaging/pull/256 replaced this PR

**Pull request closed without merging**